packages/openspp/src/Adaptor.js

```diff
@@ -660,6 +658,311 @@ export function unenroll(registrant_id, program_id) {
+    try {
+      if (typeof data !== 'object' || Array.isArray(data) || data === null) {
+        throw new Error(`${data} is not an update object!`);
+      }
+      let res = await sppConnector.searchRead(
+        'res.partner',
+        [
+          ['registrant_id', '=', group_id],
+          ['is_registrant', '=', true],
+          ['is_group', '=', true],
+        ],
+        ['id'],
+        { limit: 1, order: 'id desc' },
+      );
+      if (res.length === 0) {
+        throw new Error(`Group with registrant id: ${group_id} does not exists!`);
+      }
+      let groupId = res[0].id;
+      await sppConnector.update(
+        'res.partner',
+        groupId,
+        data
+      );
+      console.log(`ℹ Group ${group_id} updated!`);
+    } catch (err) {
+      console.log(`✗ Error: ${err}`);
+    } finally {
+      return state;
+    }
+  }
+}
+
+/**
+ * update individual for OpenSPP
+ * @public
+ * @example
+ * updateIndividual("IND_8DUQL4M4", { name: "Individual 1" })
+ * @function
+ * @param {string} individual_id - individual registrant id
+ * @param {object} data - registrant update data
+ * @returns {Operation}
+ */
+export function updateIndividual(individual_id, data) {
+  return async state => {
+    try {
+      if (typeof data !== 'object' || Array.isArray(data) || data === null) {
+        throw new Error(`${data} is not an update object!`);
+      }
+      let res = await sppConnector.searchRead(
+        'res.partner',
+        [
+          ['registrant_id', '=', individual_id],
+          ['is_registrant', '=', true],
+          ['is_group', '=', false],
+        ],
+        ['id'],
+        { limit: 1, order: 'id desc' },
+      );
+      if (res.length === 0) {
+        throw new Error(`Individual with registrant id: ${individual_id} does not exists!`);
+      }
+      let individualId = res[0].id;
+      await sppConnector.update(
+        'res.partner',
+        individualId,
+        data
+      );
+      console.log(`ℹ Individual ${individual_id} updated!`);
+    } catch (err) {
+      console.log(`✗ Error: ${err}`);
+    } finally {
+      return state;
+    }
+  }
+}
+
+/**
+ * add individual to group in OpenSPP
+ * @public
+ * @example
+ * addToGroup("GRP_B2BRHJN2", "IND_8DUQL4M4", "Head")
+ * @function
+ * @param {string} group_id - group registrant id
+ * @param {string} individual_id - individual registrant id
+ * @param {string} role - individual role in group
+ * @returns {Operation}
+ */
+export function addToGroup(group_id, individual_id, role='') {
+  return async state => {
+    try {
+      let roleId = [];
+      if (role.length > 0) {
+        roleId = await sppConnector.search(
+          'g2p.group.membership.kind',
+          [['name', '=', role]]
+        );
+        if (roleId.length === 0) {
+          roleId = [await sppConnector.create(
+            'g2p.group.membership.kind',
+            { name: role }
+          )];
+        }
+      }
+      let res = await sppConnector.searchRead(
```

Comment:
If you're not going to re-assign a variable (like `res`), you should declare it with `const`. That tells the reader that the type  (and maybe value) of the  variable won't change - which is really useful to know.

The majority of variables in modern Javascript are declared as `const` (I actually use `let` quite rarely!

Reply:
replaced